packages/typespec-client-generator-core/src/types.ts

```diff
@@ -1089,16 +1104,68 @@ export function getSdkModelPropertyType(
       })
     );
   }
-  return diagnostics.wrap({
+  const multiPartOption: MultipartOptionsType = {
+    isNameDefined: httpOperationPart ? httpOperationPart.name !== undefined : true,
+    isFilePart: isBytesInput && operationIsMultipart,
+    multi: httpOperationPart ? httpOperationPart.multi : isRawBytesArray,
+    headers: [], // TODO
+    filename: httpOperationPart?.filename
+      ? diagnostics.pipe(getSdkModelPropertyType(context, httpOperationPart.filename, operation))
+      : undefined,
+    contentType: httpOperationPart?.body.contentTypeProperty
+      ? diagnostics.pipe(
+          getSdkModelPropertyType(context, httpOperationPart.body.contentTypeProperty, operation)
+        )
+      : undefined,
+  };
+  base.isMultipartFileInput = multiPartOption.isFilePart;
+  base.multipartOptions = operationIsMultipart ? multiPartOption : undefined;
```

Comment:
Do we need to update `base.serializedName` to `httpOperationPart.name`? Are `httpOperationPart.name` the wire name?

Reply:
No, at least with model format, it is impossible to change property name with HttpPart: 
![image](https://github.com/Azure/typespec-azure/assets/70930885/4c242be4-ec21-4133-8721-0b64d9cb7086).

Reply:
I mean, which is the `name` to send in request? From the tsp I would assume it be `myPart`, not `myPropertyPart`.

Reply:
Also, in your doc this is an array without `myPropertyPart`?

![image](https://github.com/Azure/typespec-azure/assets/53292327/2a59ddb3-763b-44c9-9414-f29180e58659)

Reply:
With array format, the only way to define serialized name of part is with `#{name: "fullName"}`; with model format, there is already property name which shall be same with serialized name of part so `#{name: "fullName"}` doesn't work. @timotheeguerin please correct me if wrong.

Reply:
It is bug https://github.com/microsoft/typespec/issues/3779